src/site/topic19.rst

```diff
@@ -117,10 +117,78 @@ Factorial
 Recursive Programming
 =====================
 
-
 Factorial
 ---------
 
+* The factorial,  :math:`n!`, of a non-negative integer is the product of all non-negative between n and 1 inclusively
+
+    :math:`n! = n \times (n - 1) \times (n - 2) \times \dots \times 3 \times 2 \times 1`
+
+
+* We can nicely define this recursively
+
+.. math::
+
+    n! =
+    \begin{cases}
+        1 & \text{if $n = 0$} \\
+        n \times (n-1)! & \text{if $n > 0$} \\
+    \end{cases}
+
+
+.. note::
+
+    You will notice that :math:`0! = 1` and may wonder why. This is because:
```

Comment:
It's defined as such (retroactively) this to the [gamma function](https://en.m.wikipedia.org/wiki/Gamma_function).

Reply:
I don't think this was ultimately due to the gamma function.

Reply:
Hence (retroactively)

Reply:
Either way, I'll add another point about it aligning with gamma function, which will give a chance to talk about gamma function, which is cool n' all.

Reply:
Gamma might be a little too advanced for first years. If I remember correctly, it was a latter half third year pure math course in which I was introduced to it.

Reply:
You're absolutely right, but I'll just leave it as a link.